### PR TITLE
Start a Workflow Execution from the UI

### LINKED ISSUE #1525: Add ability to trigger workflow from web-ui
## Is your feature request related to a problem? Please describe.
No. 
The feature request stems from convenience. It takes time to connect locally to the Temporal front-end. If I have an open connection to a web-ui, and I see a job has an issue or I need to start a job, it would be convenient to trigger the job from the web-ui. This would also allow jobs to be triggered in different environments more easily. 

## Describe the solution you'd like

It would be great if in addition to "pause", the scheduler screen had an option to 'start' or 'trigger once' selected workflows. This option would run the workflow once starting now. Similar to triggering the job from the command line `temporal workflow start <options>`

Screenshot 2023-08-02 at 11 37 43 AM

## Describe alternatives you've considered

This could also be handled from the main workflow screen, but only if that screen showed not just running workflows, but all available workflows. 

## Additional context

Yes. I am just being lazy. ¯\_(ツ)_/¯ 
Thanks for all the work improving Temporal. It's a great service.

### PR DESCRIPTION
**Pull request by Alex-Tideman**
## Description & motivation 💭 
Allow a user to start a Workflow Execution from the UI.

A user can start with a blank template by clicking on the primary Start Workflow button on the Workflow List page, or they can start with a prefilled template from a Workflow List row or Workflow Execution Page.

**To enable Start Workflow on UI:**
TEMPORAL_START_WORKFLOW_DISABLED = "false"
By default, this is set to true except in development.

**Required:**
- A user must enter or generate a WorkflowID
- A user must enter a Task Queue. They can click 'Check Status' to view how many workers are currently running on the Task Queue and a link to the Task Queue page.
- A user must enter a WorkflowType.

On blur of any of these inputs, the url searchParams will update with the values so refreshing or sharing the link will preserve the values for quick creation.

**Optional:**

- A user can enter an input, or they can upload a .json file, or they can fetch the previous input of a Workflow with the same WorkflowID & WorkflowType.
- A user can add one or more Custom Search Attributes.

All buttons to navigation to Start a Workflow are hidden if StartWorkflowDisabled is true from Settings API. The Start Workflow submit button is hidden if namespaceWrite permissions are disabled or disableWriteActions is set to true.

### Screenshots (if applicable) 📸 
Screen Shot 2024-05-15 at 2 22 26 PM
Screen Shot 2024-05-15 at 2 32 34 PM
Screen Shot 2024-05-15 at 2 22 37 PM
Screen Shot 2024-05-15 at 2 33 19 PM
Screen Shot 2024-05-15 at 2 33 32 PM
Screen Shot 2024-05-15 at 2 33 55 PM

### How was this tested 👻 

- [x] Manual testing
- [x] E2E tests added
- [ ] Unit tests added

### Draft Checklist 

- [ ] Figure out how to format DateTime and KeywordList SAs to submit them correctly
- [x] Fix lame types on SA Inputs
- [x] Confirm label text and translate

### Merge Checklist 

- [x] E2E tests
- [x] Test input with codec server

### Issue(s) closed 

Closes #1525

### CODE REVIEW

File: src/lib/components/workflow/start-workflow-button.svelte [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    on:click={() => goto(href)}
+    {...$$restProps}
+  >
+    <Icon title="Start Workflow" name="lightning-bolt" />
+  </button>
+</Tooltip>
+
+<style lang="postcss">
+  .start-button {
+    @apply m-1 rounded-md border-2 border-[transparent] bg-transparent p-1 hover:surface-interactive-secondary focus-visible:surface-interactive-secondary focus-visible:border-indigo-600 focus-visible:outline-none;
```
Comment (stevekinney) [inline src/lib/components/workflow/start-workflow-button.svelte:38]: `tailwind-merge` really doesn't like it when you mix in other classes. So, I'd recommend just putting these classes into the class string on Line 28.

It _does_ beg the question—why not just use a Holocene button?

File: src/lib/components/workflow/start-workflow-input-upload.svelte [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+            variant: 'error',
+            message: translate('events.event-history-load-error'),
+          });
+        }
+      };
+    }
+  };
+</script>
+
+<label for="start-workflow-input-file-upload" class="file-upload-label">
```
Comment (stevekinney) [inline src/lib/components/workflow/start-workflow-input-upload.svelte:29]: Should we use the [`Label`](https://github.com/temporalio/ui/blob/fix-invisible-drawer-close-icon/src/lib/holocene/label.svelte) component?

File: src/lib/components/workflow/start-workflow-input-upload.svelte [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  <Icon name="upload" />
+</label>
+<input
```
Comment (stevekinney) [inline src/lib/components/workflow/start-workflow-input-upload.svelte:32]: Not important for now, but we probably want to make a File Upload component in Holocene at some point.

File: src/lib/components/workflow/start-workflow-input-upload.svelte [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+  id="start-workflow-input-file-upload"
+  class="block w-10 rounded-full border border-slate-200 p-2"
+  type="file"
+  accept=".json"
+  on:change={onFileSelect}
+/>
+
+<style lang="postcss">
+  input[type='file'] {
```
Comment (stevekinney) [inline src/lib/components/workflow/start-workflow-input-upload.svelte:41]: Why not just give the input a `hidden` class and call it a day?

File: src/lib/components/workflow/start-workflow-input-upload.svelte [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+    display: none;
+  }
+
+  .file-upload-label {
```
Comment (stevekinney) [inline src/lib/components/workflow/start-workflow-input-upload.svelte:45]: Ah, the label isn't a label, it's a button?

File: src/lib/components/workflow/start-workflow-button.svelte (hunk 1 above)
Comment (Alex-Tideman) [reply src/lib/components/workflow/start-workflow-button.svelte:40]: Remove the merge, it's not needed. It's not a Holocene button cuz those are too big for table rows. Definitely can add another variant if we want.

File: src/lib/components/workflow/start-workflow-input-upload.svelte (hunk 3 above)
Comment (Alex-Tideman) [reply src/lib/components/workflow/start-workflow-input-upload.svelte:32]: Yeah totally should have this be a Holocene component if this is the design we want for all file uploads.

File: src/lib/components/workflow/start-workflow-input-upload.svelte (hunk 4 above)
Comment (Alex-Tideman) [reply src/lib/components/workflow/start-workflow-input-upload.svelte]: yupppp

File: src/lib/components/workflow/start-workflow-search-attribute-input.svelte [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+  export let searchAttributes: SearchAttributeOption[] = [];
+  export let attribute: SearchAttributeInput;
+  export let onRemove: (attribute: string) => void;
+
+  $: type =
+    searchAttributes.find((a) => a.value === attribute.attribute)?.type ?? '';
+
+  const clearValue = (_) => (attribute.value = '');
+
+  $: clearValue(type);
```
Comment (stevekinney) [inline src/lib/components/workflow/start-workflow-search-attribute-input.svelte:22]: I had to squint to figure out what you're doing here, but it seems a bit too clever.

File: src/lib/components/workflow/start-workflow-button.svelte (hunk 1 above)
Comment (rossedfort) [reply src/lib/components/workflow/start-workflow-button.svelte:40]: We have a `size` prop on button that should do the trick. If `xs` is too small, we could add `xxs`

File: src/lib/holocene/file-input.svelte [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+</script>
+
+<label for={id} class="file-upload-label">
+  <Icon name="upload" />
+</label>
+<input {id} class="hidden" type="file" {accept} on:change={onFileSelect} />
+
+<style lang="postcss">
+  .file-upload-label {
+    @apply relative flex h-10 w-fit cursor-pointer items-center justify-center rounded-lg border border-secondary px-4 py-2 text-base text-primary transition-shadow hover:surface-interactive-secondary focus-visible:surface-interactive-secondary focus-visible:border-white focus-visible:outline-none disabled:cursor-not-allowed disabled:opacity-50 dark:hover:border-transparent dark:focus-visible:border-black;
```
Comment (laurakwhit) [inline src/lib/holocene/file-input.svelte:38]: What do you think of using similar styling to the secondary `Button` component here? E.g. ⬇️ 

```suggestion
    @apply relative flex h-10 w-fit cursor-pointer items-center justify-center rounded-lg border border-secondary px-4 py-2 text-base text-primary surface-primary border-subtle focus-visible:ring-primary/70 hover:surface-interactive-secondary focus-visible:surface-interactive-secondary;
```

File: src/lib/pages/start-workflow.svelte [hunk 8]
```diff
@@ -0,0 +1,269 @@
...
+  $: enableStart =
+    !!workflowId &&
+    !!taskQueue &&
+    !!workflowType &&
+    !workflowCreateDisabled($page);
+
+  $: checkTaskQueue(taskQueueParam);
+</script>
+
+<div class="flex w-full flex-col items-center pb-24">
```
Comment (laurakwhit) [inline src/lib/pages/start-workflow.svelte:151]: Do we want to add any kind of breadcrumbs here?

Screenshot 2024-06-24 at 11 13 59 AM

File: src/lib/pages/start-workflow.svelte [hunk 9]
```diff
@@ -0,0 +1,269 @@
...
+      allowEmpty: true,
+    });
+  };
+
+  const onUpload = (uploadInput: string) => {
+    input = uploadInput;
+    inputRetrieved = Date.now();
+  };
+
+  $: enableStart =
```
Comment (laurakwhit) [inline src/lib/pages/start-workflow.svelte:142]: Do we want to check `pollerCount` here as well? Or is it fine to start even if Task Queue is inactive?

Screenshot 2024-06-24 at 11 13 42 AM

File: src/lib/components/workflow/start-workflow-search-attributes.svelte [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+    {searchAttributes}
+    bind:attribute
+    {onRemove}
+  />
+{/each}
+<Button
+  variant="ghost"
+  leadingIcon="add"
+  on:click={addSearchAttribute}
+  disabled={!searchAttributes.length ||
```
Comment (laurakwhit) [inline src/lib/components/workflow/start-workflow-search-attributes.svelte:37]: Wondering if we might want some kind of tooltip when there are no search attributes and this is disabled?

Screenshot 2024-06-24 at 11 12 25 AM

File: src/lib/components/workflow/start-workflow-search-attribute-input.svelte [hunk 11]
```diff
@@ -0,0 +1,69 @@
...
+      label="Value"
+      id="attribute-value"
+      class="w-full"
+      bind:value={attribute.value}
+    />
+  {/if}
+  <Button
+    variant="secondary"
+    leadingIcon="close"
+    class="mt-6"
```
Comment (laurakwhit) [inline src/lib/components/workflow/start-workflow-search-attribute-input.svelte:66]: What do you think about making this close button a little more similar to close buttons elsewhere?

```suggestion
    variant="ghost"
    leadingIcon="close"
    class="mt-6 w-10 rounded-full"
```

| Before | After |
|--|--|
|Screenshot 2024-06-24 at 12 23 10 PM|Screenshot 2024-06-24 at 12 22 47 PM|

File: src/lib/components/workflow-actions.svelte [hunk 12]
```diff
@@ -138,6 +141,49 @@
...
+          routeForWorkflowStart({
+            namespace,
+            workflowId: workflow.id,
+            taskQueue: workflow.taskQueue,
+            workflowType: workflow.name,
+          }),
+        )}
+      data-testid="start-workflow-button"
+    >
+      <div class="flex grow items-center gap-1">
```
[the remark this answers is not in the document]
Comment (Alex-Tideman) [reply src/lib/components/workflow-actions.svelte]: 👍

File: src/lib/components/workflow-actions.svelte [hunk 13]
```diff
@@ -138,6 +141,49 @@
...
+          routeForWorkflowStart({
+            namespace,
+            workflowId: workflow.id,
+            taskQueue: workflow.taskQueue,
+            workflowType: workflow.name,
+          }),
+        )}
+      data-testid="start-workflow-button"
+    >
+      <div class="flex items-center gap-1">
```
[the remark this answers is not in the document]
Comment (Alex-Tideman) [reply src/lib/components/workflow-actions.svelte]: 👍 

I tried making that a one-liner, but it ends up needing more finagling on the menu-item than I liked. Can def revisit that.

File: src/lib/components/workflow/start-workflow-button.svelte [hunk 14]
```diff
@@ -0,0 +1,36 @@
...
+    size="xs"
+    variant="ghost"
+    class="start-button"
+    on:click={() => goto(href)}
+    {...$$restProps}
+  >
+    <Icon
+      title={translate('workflows.start-workflow-like-this-one')}
+      name="lightning-bolt"
+    />
```
[the remark this answers is not in the document]
Comment (Alex-Tideman) [reply src/lib/components/workflow/start-workflow-button.svelte]: 👍

File: src/lib/holocene/file-input.svelte (hunk 7 above)
Comment (Alex-Tideman) [reply src/lib/holocene/file-input.svelte]: 👍

File: src/lib/pages/start-workflow.svelte (hunk 8 above)
Comment (Alex-Tideman) [reply src/lib/pages/start-workflow.svelte:152]: 👍

File: src/lib/pages/start-workflow.svelte (hunk 9 above)
Comment (Alex-Tideman) [reply src/lib/pages/start-workflow.svelte:143]: I think we want to still allow it (maybe worker is starting later and they don't care), but good call out. We can see what users think

File: src/lib/components/workflow/start-workflow-search-attribute-input.svelte (hunk 11 above)
Comment (Alex-Tideman) [reply src/lib/components/workflow/start-workflow-search-attribute-input.svelte]: 👍

File: src/lib/components/workflow/start-workflow-search-attributes.svelte (hunk 10 above)
Comment (Alex-Tideman) [reply src/lib/components/workflow/start-workflow-search-attributes.svelte:37]: Great idea, but not gonna do it at the moment. I think we'll get some more feedback on SAs in general and will make a bigger change.

File: src/lib/components/workflow/start-workflow-search-attribute-input.svelte [hunk 15]
```diff
@@ -0,0 +1,69 @@
...
+  {:else}
+    <Input
+      label="Value"
+      id="attribute-value"
+      class="w-full"
+      bind:value={attribute.value}
+    />
+  {/if}
+  <Button
+    variant="secondary"
```
Comment (laurakwhit) [inline src/lib/components/workflow/start-workflow-search-attribute-input.svelte:64]:
```suggestion
    variant="ghost"
```

**Review by laurakwhit — APPROVED** [review]
One more small suggestion, but otherwise lgtm 🚀

File: src/lib/components/workflow/start-workflow-search-attribute-input.svelte (hunk 15 above)
Comment (Alex-Tideman) [reply src/lib/components/workflow/start-workflow-search-attribute-input.svelte]: 👍

**Pull request merged**